django/core/mail/message.py

```diff
@@ -309,7 +309,11 @@ def attach(self, filename=None, content=None, mimetype=None):
     def attach_file(self, path, mimetype=None):
         """Attaches a file from the filesystem."""
         filename = os.path.basename(path)
-        with open(path, 'rb') as f:
+        if not mimetype:
+            mimetype, _ = mimetypes.guess_type(filename)
```

Comment:
I think we should have a test and handling for the case where `mimetypes.guess_type()` returns `None` as done in `_create_attachment()`.